h/static/scripts/annotation-mapper.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+
+// Fetch the container object for the passed annotation from the threading
```

Comment:
It's not in the docs. The threading service is a wrapper around [this implementation](https://github.com/maxogden/conversationThreading-js) of the [venerable JWZ threading algorithm](https://www.jwz.org/doc/threading.html).

I will note, however, that I haven't changed any of that in this pull request. This is a straight translation of CoffeeScript to JavaScript, and this is a small helper function to make the code easier to read.